### PR TITLE
Added option to select encoding of .bib file in parse_file()

### PR DESCRIPTION
**Pull request by himcraft**

### CODE REVIEW

**Comment by MiWeiss** [comment]
Thanks @himcraft for the PR. 

The code change looks good, but would you mind adding a few tests? I know that the change is rather trivial, thus the purposes of the tests would not be to verify if you change is correct (I am convinced that it is), but instead serve to prevent from regression, i.e., from unintentionally breaking this function in the future.

Thanks!

**Comment by himcraft** [comment]
Added `test_encoding.py` and a `gbk_test.bib` file in `tests/`.
This is my first attempt at writing tests for a Python project. Please feel free to point out any errors or suggest improvements :)

File: tests/test_encoding.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+"""Testing the parse_file function."""
+
+import pytest
+
+from bibtexparser import parse_file, writer
+
+
+def test_gbk():
+    library = parse_file("tests/gbk_test.bib", encoding="gbk")
+    lines = writer.write(library).splitlines()
```
Comment (MiWeiss) [inline tests/test_encoding.py:10]: I would suggest not making the assertion on the "re-written" bibtex, but on the parsed entities instead.

e.g. `assert library.entries[0]["author"] == ...`

That way, we can keep the test independent of the writer.

File: tests/test_encoding.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+"""Testing the parse_file function."""
```
Comment (MiWeiss) [inline tests/test_encoding.py:1]: can we re-name this file to `test_entrypoint` and the module docstring accordingly - there's likely to be added more tests to it eventually.

File: tests/gbk_test.bib [hunk 3]
```diff
@@ -0,0 +1,6 @@
+@ARTICLE{����2013,
```
Comment (MiWeiss) [inline tests/gbk_test.bib:1]: Please move this file into a (new) `tests/resources` folder.

**Review by MiWeiss — CHANGES_REQUESTED** [review]
Thanks a lot for your commit.

Your test would certainly already be sufficient as a regression test. I still have left some minor comments, mostly to keep the test atomic and the test organisation (read file structure) consistent with the remainder of the test suite.

Afterwards, this will be ready to be merged :rocket: Thanks!

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -28,5 +28,5 @@ Issues are labelled `v1` and `v2`, correspondingly.
 To install the dev dependencies, run `pip install -e .[test,lint,docs]` from within the cloned repository. Then:
 
 - To test your code, run `pytest .`
-- To lint your code (required for CI/CD to pass), run: `black bibtexparser tests docs && isort bibtexparser tests docs --profile black
+- To lint your code (required for CI/CD to pass), run: `black bibtexparser tests docs && isort bibtexparser tests docs --profile black`
```
Comment (MiWeiss) [inline CONTRIBUTING.md:31]: Nice catch ;-)

**Review by MiWeiss — APPROVED** [review]
lgtm!

Thanks a lot for your contribution. If there's any more problems or missing features you spot, please do not hesitate to report them.

**Pull request merged**